examples/twitter-compose-with-typeahead/src/Autocomplete.tsx

```diff
@@ -0,0 +1,287 @@
+                state.collections.map(({ source, items }) => {
+                  return (
+                    <div
+                      key={`source-${source.sourceId}`}
+                      className={[
+                        'autocomplete-source',
+                        state.status === 'stalled' &&
+                          'autocomplete-source-stalled',
+                      ]
+                        .filter(Boolean)
+                        .join(' ')}
+                    >
+                      {items.length > 0 && (
+                        <ul
+                          {...autocomplete.getListProps()}
+                          className="autocomplete-items"
+                        >
+                          {items.map((item) => {
+                            const itemProps = autocomplete.getItemProps({
+                              item,
+                              source,
+                            });
+
+                            return (
+                              <li key={item.handle} {...itemProps}>
+                                <div
+                                  className={[
+                                    'autocomplete-item',
+                                    itemProps['aria-selected'] &&
+                                      'autocomplete-item-selected',
+                                  ]
+                                    .filter(Boolean)
+                                    .join(' ')}
+                                >
+                                  <AccountItem hit={item} />
+                                </div>
+                              </li>
+                            );
+                          })}
+                        </ul>
+                      )}
+                    </div>
+                  );
+                })}
+            </div>
+          </div>
+        </div>
+        <div className="box-footer">
+          <button type="submit" className="tweet-button">
+            Tweet
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+type AccountItemProps = {
+  hit: Hit<Account>;
+};
+
+function AccountItem({ hit }: AccountItemProps) {
+  return (
+    <>
+      {(hit.follows_me || hit.followed_by_me) && (
+        <div className="account-follow-status">
+          <svg
+            className="account-follow-status-icon"
+            viewBox="0 0 20 20"
+            fill="currentColor"
+          >
+            <path
+              fillRule="evenodd"
+              d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
+              clipRule="evenodd"
+            />
+          </svg>
+          {hit.follows_me && hit.followed_by_me && (
+            <span>You follow each other</span>
+          )}
+          {!hit.follows_me && hit.followed_by_me && (
+            <span>You follow them</span>
+          )}
+          {hit.follows_me && !hit.followed_by_me && (
+            <span>They follow you</span>
+          )}
```

Comment:
This feels like noise. Do we really need this in the example?

Reply:
This is used in custom ranking, so this is helpful to understand relevance. We can move it into a component to streamline `Autocomplete`, wdyt?

Reply:
I agree with Francois, the example is about rendering, but the following information doesn't really matter for creating let's say a mention inside google docs

Reply:
I agree on the fact that it adds noise, and initially I wanted to drop them. However, the initial tutorial covers the indexing and index settings. Would you drop it in this new one?

If yes, it does make sense to drop the code; otherwise it would feel lacking.

Reply:
I think so. I didn't read the guide yet, wanted to validate the example first.

Reply:
Handled in 094b7bc.